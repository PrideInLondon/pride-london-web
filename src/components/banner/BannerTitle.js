import React from 'react'
import PropTypes from 'prop-types'
import { StyledBannerTitle, BannerTitleWrapper } from './BannerTitle.styles'

const BannerTitle = ({ children }) => (
  <BannerTitleWrapper>
    <StyledBannerTitle>{children}</StyledBannerTitle>
  </BannerTitleWrapper>
)

BannerTitle.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
}

export default BannerTitle
