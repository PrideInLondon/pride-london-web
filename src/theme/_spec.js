import { screenSizes, breakpoints } from './theme'

it.each`
  screen         | breakpointIndex
  ${'mobile'}    | ${0}
  ${'tablet'}    | ${1}
  ${'desktop'}   | ${2}
  ${'desktopHD'} | ${3}
`('successfully assigns $screen screen size', ({ screen, breakpointIndex }) => {
  expect(screenSizes[screen]).toBe(breakpoints[breakpointIndex])
})
