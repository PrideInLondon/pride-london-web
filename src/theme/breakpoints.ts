export const sm = 375
export const md = 768
export const lg = 1024
export const xl = 1440
export const nav = 1360

export interface Breakpoints {
  sm: string
  md: string
  lg: string
  xl: string
  nav: string
}

export const breakpoints: string[] & Breakpoints = Object.assign(
  [`${sm}px`, `${md}px`, `${lg}px`, `${xl}px`],
  {
    sm: `${sm}px`,
    md: `${md}px`,
    lg: `${lg}px`,
    xl: `${xl}px`,
    nav: `${nav}px`,
  }
)
