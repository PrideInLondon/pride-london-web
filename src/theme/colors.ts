export const colors = {
  black: '#000000',
  darkGrey: '#5B5B5B',
  mediumGrey: '#E1E1E1',
  lightGrey: '#EFEFEF',
  white: '#FFFFFF',
  yellow: '#FFD95E',
  tomato: '#FF5B44',
  pink: '#ED2C6E',
  fuscia: '#FF88F1',
  electricPurple: '#BC01FF',
  darkPurple: '#660C63',
  indigo: '#2D2F7F',
  darkIndigo: '#20215A',
  greyBlue: '#5770D1',
  beachBlue: '#2C77E5',
  skyBlue: '#2ACFFF',
  bondiBlue: '#00A3BD',
  darkCyan: '#008296',
  eucalyptusGreen: '#2CDA9D',
  lemonGreen: '#C3D30E',
  green: '#34C30A',
  amber: '#FFCA00',
  red: '#D0021B',
  gold: '#FFE16C',
  silver: '#E0DFDC',
  bronze: '#B59035',
  mexicanPink: '#E6007E',
  raisinBlack: '#272727',
}
