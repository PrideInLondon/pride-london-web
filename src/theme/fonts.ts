export const fonts = {
  title: 'Poppins, sans-serif',
  body: 'Roboto, sans-serif',
}
