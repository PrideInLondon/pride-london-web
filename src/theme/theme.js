const theme = {
  colors: {
    black: '#000000',
    darkGrey: '#5B5B5B',
    mediumGrey: '#E1E1E1',
    lightGrey: '#EFEFEF',
    white: '#FFFFFF',
    yellow: '#FFD95E',
    tomato: '#FF5B44',
    pink: '#ED2C6E',
    fuscia: '#FF88F1',
    electricPurple: '#BC01FF',
    darkPurple: '#660C63',
    indigo: '#2D2F7F',
    greyBlue: '#5770D1',
    beachBlue: '#2C77E5',
    skyBlue: '#2ACFFF',
    bondiBlue: '#00A3BD',
    eucalyptusGreen: '#2CDA9D',
    lemonGreen: '#C3D30E',
    green: '#34C30A',
    amber: '#FFCA00',
    red: '#D0021B',
  },
  fonts: {
    title: 'Poppins, sans-serif',
    body: 'Roboto, sans-serif',
  },
  space: [0, 5, 10, 15, 20],
  breakpoints: [375, 768, 1024, 1440],
  navBreakpoint: 1275,
}

export default theme
