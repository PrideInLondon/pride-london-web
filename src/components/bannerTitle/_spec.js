import React from 'react'
import { shallow } from 'enzyme'
import theme from '../../theme/theme'
import BannerTitle from './'
import 'jest-styled-components'

describe('BannerTitle', () => {
  it('renders', () => {
    const tree = shallow(<BannerTitle>foo</BannerTitle>)
    expect(tree).toMatchSnapshot()
  })

  it('renders a <h1 />', () => {
    const wrapper = shallow(<BannerTitle>foo</BannerTitle>)
    expect(wrapper.find('TestBannerTitle')).toHaveLength(1)
  })

  it('renders child text ', () => {
    const text = 'Here is a test title!'
    const wrapper = shallow(
      <BannerTitle>
        {text}
      </BannerTitle>, {
      context: { theme },
    })
    // Currently having issues testing with enzyme as the theme does not get provided to the styled componentw
    expect(wrapper).toMatchSnapshot()
  })
})
