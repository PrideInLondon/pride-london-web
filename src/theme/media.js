import { css } from 'styled-components'
import theme from './theme'

export const media = {
  // Min width media queries
  mobile: (...args) => css`
    @media (min-width: ${theme.breakpoints[0]}px) {
      ${css(...args)};
    }
  `,
  tablet: (...args) => css`
    @media (min-width: ${theme.breakpoints[1]}px) {
      ${css(...args)};
    }
  `,
  desktop: (...args) => css`
    @media (min-width: ${theme.breakpoints[2]}px) {
      ${css(...args)};
    }
  `,

  desktopHD: (...args) => css`
    @media (min-width: ${theme.breakpoints[3]}px) {
      ${css(...args)};
    }
  `,
  // Custom breakpoint for navigation
  nav: (...args) => css`
    @media (min-width: ${theme.navBreakpoint}px) {
      ${css(...args)};
    }
  `,

  // Max width media queries
  mobileMax: (...args) => css`
    @media (max-width: ${theme.breakpoints[0] - 1}px) {
      ${css(...args)};
    }
  `,
  tabletMax: (...args) => css`
    @media (max-width: ${theme.breakpoints[1] - 1}px) {
      ${css(...args)};
    }
  `,
  desktopMax: (...args) => css`
    @media (max-width: ${theme.breakpoints[2] - 1}px) {
      ${css(...args)};
    }
  `,

  desktopHDMax: (...args) => css`
    @media (max-width: ${theme.breakpoints[3] - 1}px) {
      ${css(...args)};
    }
  `,
  // Custom breakpoint for navigation
  navMax: (...args) => css`
    @media (max-width: ${theme.navBreakpoint - 1}px) {
      ${css(...args)};
    }
  `,
}
