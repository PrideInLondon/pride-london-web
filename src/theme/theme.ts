import { breakpoints } from './breakpoints'
import { mediaQueries } from './mediaQueries'
import { colors } from './colors'

const theme = {
  colors,
  fonts: {
    title: 'Poppins, sans-serif',
    body: 'Roboto, sans-serif',
  },
  space: [0, 5, 10, 15, 20],
  breakpoints,
  mediaQueries,
  navBreakpoint: 1360,
}

export default theme
