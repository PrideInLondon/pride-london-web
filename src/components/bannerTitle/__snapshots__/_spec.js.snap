// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerTitle renders 1`] = `
<bannerTitle__BannerTitleWrapper>
  <bannerTitle__BannerTitle>
    foo
  </bannerTitle__BannerTitle>
</bannerTitle__BannerTitleWrapper>
`;

exports[`BannerTitle renders child text  1`] = `
<bannerTitle__BannerTitleWrapper>
  <bannerTitle__BannerTitle>
    Here is a test title!
  </bannerTitle__BannerTitle>
</bannerTitle__BannerTitleWrapper>
`;
